bin/check-a11y.js

```diff
@@ -23,102 +23,181 @@ const pkgFilter = hasFilter
 }, [])
 
-let issues = {
-  minor: 0,
-  moderate: 0,
-  serious: 0,
-  critical: 0,
+const mergeResultArrays = (newArray, originArray = []) => {
+  const mergedNodes = (a, b = []) =>
+    [...a, ...b].reduce((acc, curr) => {
+      const { target, ...rest } = curr
+
+      const mergedNode = {
+        ...rest,
+        target: [
+          ...new Set([...acc.reduce((prev, curr) => [...prev, ...curr.target], []), ...target]),
+        ],
+      }
+
+      return [...acc, mergedNode]
+    }, [])
+
+  return [...originArray, ...newArray].reduce((acc, currentRule) => {
+    const duplicateRule = acc.filter(item => item.id === currentRule.id)
+
+    if (duplicateRule.length > 0) {
+      const uniqRules = acc.filter(item => item.id !== currentRule.id)
+
+      const duplicateRuleNodes = duplicateRule.reduce((acc, curr) => {
+        const { nodes, ...rest } = curr
+        return [...acc, ...nodes]
+      }, [])
+
+      const nodes = mergedNodes(duplicateRuleNodes, currentRule.nodes)
+
+      return [
+        ...uniqRules,
+        {
+          ...currentRule,
+          nodes: [nodes[nodes.length - 1]],
+        },
+      ]
+    } else {
+      const nodes = mergedNodes(currentRule.nodes)
+
+      return [
+        ...acc,
+        {
+          ...currentRule,
+          nodes: [nodes[nodes.length - 1]],
+        },
+      ]
+    }
+  }, [])
 }
 
+let report = {}
+
+/**
+ *
+ * Informations about Axe API available here:
+ * https://github.com/dequelabs/axe-core/blob/master/doc/API.md
+ */
 const checkA11y = async () => {
   const driver = new Builder()
     .forBrowser('chrome')
     .setChromeOptions(new Options().addArguments('-headless'))
     .build()
 
-  let testedStory
+  let testedComponent
 
-  for (const storyId of storiesList) {
-    if (testedStory !== stories[storyId].title)
-      console.log(`Checking accessibility for ${stories[storyId].title}...`)
+  for (const docId of docsList) {
+    if (testedComponent !== entries[docId].title) {
+      console.log(`Checking accessibility for ${entries[docId].title}...`)
+    }
 
-    testedStory = stories[storyId].title
+    testedComponent = entries[docId].title.toLowerCase()
 
     try {
-      await driver.get(`http://localhost:6006/iframe.html?viewMode=story&id=${storyId}`)
+      await driver.get(`http://localhost:6006/?path=/docs/${docId}`)
 
       const results = await new AxeBuilder(driver)
         .options({
           resultTypes: ['violations', 'incomplete'],
           rules: {
+            // 'meta-viewport': { enabled: false },
             'page-has-heading-one': { enabled: false },
             'landmark-one-main': { enabled: false },
             region: { enabled: false },
             'color-contrast': { enabled: false },
           },
           runOnly: ['best-practice', 'wcag2a', 'wcag2aa', 'wcag21a', 'wcag21aa'],
-          reporter: 'no-passes',
+          reporter: 'v2',
         })
-        .exclude('.docblock-argstable')
+        .exclude('*:not(.docs-story)')
```

Comment:
Working with stories was not relevant anymore, and was causing some issues. We prefer now to parse the complete documentation page and define which DOM nodes we want to check.